src/s1reader/s1_orbit.py

```diff
@@ -56,16 +61,69 @@ def download_orbit(safe_file: str, orbit_dir: str):
     # Find precise orbit first
     orbit_dict = get_orbit_dict(mission_id, start_time,
                                 end_time, 'AUX_POEORB')
+
     # If orbit dict is empty, find restituted orbits
     if orbit_dict is None:
         orbit_dict = get_orbit_dict(mission_id, start_time,
                                     end_time, 'AUX_RESORB')
+
     # Download orbit file
-    orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
-    if not os.path.exists(orbit_file):
-        download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
+    if orbit_dict is not None:
+        orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
+        if not os.path.exists(orbit_file):
+            download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
 
-    return orbit_file
+        return orbit_file
+
+    # POEORB is not found, or there is no RESORB file that
+    # covers the sensing period + margin at the starting time.
+    # Try to find two subsequent RESORB files that covers the
+    # sensing period + margins at the starting time.
+    if orbit_dict is None:
+        pad_short = datetime.timedelta(seconds = PADDING_SHORT)
+        print('Attempting to download RESORB files.')
+        resorb_dict_earlier = get_orbit_dict(mission_id,
+                                            start_time,
+                                            start_time + 2 * pad_short, 'AUX_RESORB')
+
+        resorb_dict_later = get_orbit_dict(mission_id,
+                                          start_time + margin_start_time - pad_short,
+                                          end_time + pad_short,
+                                          'AUX_RESORB')
+
+        resorb_dict_list = [resorb_dict_earlier, resorb_dict_later]
+
+        if all(resorb_dict_list):
```

Comment:
This will simplify your code:

```suggestion
        if not all(resorb_dict_list):
            warnings.warn('Cannot find RESORB pair that covers the timeframe in the query')
            return
```

And indent the rest of the block to the left.

Reply:
good point. we can do early return here.